File: src/main/java/bad/robot/radiate/ui/FadeOut.java
package bad.robot.radiate.ui;

import java.beans.PropertyChangeListener;

import static java.lang.String.format;
import static java.util.Arrays.stream;

@Deprecated
class FadeOut implements Fade {

    private float limit = 10f;
    private float count = limit;

    @Override
    public void fireEvent(PropertyChangeListener... listeners) {
        if (--count >= 0) {
            stream(listeners).forEach(listener -> listener.propertyChange(new AlphaTransparencyChangeEvent(this, (count + 1) / limit, count / limit)));
        }
    }

    @Override
    public boolean done() {
        return count == 0;
    }

    @Override
    public String toString() {
        return format("%.2f", count / limit);
    }
}

File: src/main/scala/bad/robot/radiate/ui/FadeOut.scala
package bad.robot.radiate.ui

import java.beans.PropertyChangeListener

class FadeOut extends Fade {

  private val limit = 10f

  private var count = limit

  def fireEvent(listeners: List[PropertyChangeListener]) {
    count -= 1
    if (count >= 0) {
      listeners.foreach(_.propertyChange(new AlphaTransparencyChangeEvent(this, (count + 1) / limit, count / limit)))
    }
  }

  def done = count == 0

  override def toString = f"${count / limit}%.2f"
}